### PR TITLE
Genealogy: Swapping to Object Reference

### PR DESCRIPTION
**Pull request by Windchild292**
This PR swaps Genealogy and FormerSpouse to using Object references and moves FormerSpouseReason into an enum. The changes have been tested using the campaigns located in #1765

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/personnel/familyTree/Genealogy.java [hunk 1]
```diff
@@ -264,9 +257,9 @@ public boolean checkMutualAncestors(Person person, Campaign campaign) {
             // If so, decrease remaining search depth
             depth--;
             // Then parse through the parents
-            for (UUID parent : getParents()) {
+            for (Person parent : getParents()) {
                 // And add all of their returned ancestors to the list
-                ancestors.addAll(campaign.getPerson(parent).getGenealogy().getAncestors(depth, campaign));
+                ancestors.addAll(parent.getGenealogy().getAncestors(depth));
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/personnel/familyTree/Genealogy.java:262]: Is a call to getGenealogy() guaranteed to return non-null?

File: MekHQ/src/mekhq/campaign/personnel/familyTree/Genealogy.java [hunk 2]
```diff
@@ -118,60 +108,64 @@ public void setSpouse(@Nullable UUID spouse) {
...
      * This is used to add a new family member
      * @param relationshipType the relationship type between the two people
-     * @param id the id of the person to add
+     * @param person the person to add
      */
-    public void addFamilyMember(FamilialRelationshipType relationshipType, UUID id) {
-        if (id != null) {
+    public void addFamilyMember(FamilialRelationshipType relationshipType, @Nullable Person person) {
+        if (person != null) {
             getFamily().putIfAbsent(relationshipType, new ArrayList<>());
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/personnel/familyTree/Genealogy.java:142]: Is getFamily() guaranteed to be non-null?

File: MekHQ/src/mekhq/campaign/personnel/familyTree/Genealogy.java [hunk 3]
```diff
@@ -223,7 +217,7 @@ public boolean hasParents() {
      * @return true if they have mutual ancestors, otherwise false
      */
     public boolean checkMutualAncestors(Person person, Campaign campaign) {
-        if (getOrigin().equals(person.getId())) {
+        if (getOrigin().equals(person)) {
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/personnel/familyTree/Genealogy.java:220]: Is getOrigin() guaranteed to be non-null?

**Review by NickAragua — COMMENTED** [review]
A few questions about whether things that are assumed to be non-nullable are actually going to be non-nullable. I definitely didn't spot all instances where you're referencing things that look like they may come back as null, but I think it's worth going through this file and checking those calls.

File: MekHQ/src/mekhq/campaign/personnel/familyTree/Genealogy.java (hunk 3 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/personnel/familyTree/Genealogy.java:220]: It wasn't only for a few error conditions, but that will be fixed... swapped around a few XML bits to improve and fix that potential issue.

File: MekHQ/src/mekhq/campaign/personnel/familyTree/Genealogy.java (hunk 1 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/personnel/familyTree/Genealogy.java:262]: It is. Anything that is nullable is marked as such.

File: MekHQ/src/mekhq/campaign/personnel/familyTree/Genealogy.java (hunk 2 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/personnel/familyTree/Genealogy.java:142]: It is. Anything that is nullable is marked as such.

**Comment by Windchild292** [comment]
Reworked a bunch of loading to ensure the origin cannot be null and tested with loading my 0.46.1, 0.47.7, 0.47.11, and 0.49.0 test games.

**Review by NickAragua — APPROVED** [review]

**Pull request merged**